src/packet_router/login_init.rs

```diff
@@ -14,25 +13,73 @@ pub fn init_login(
 ) {
     println!("Login protocol initiated :{:?}", p);
     match p.clone() {
-        Packet::LoginStart(packet) => {
-            println!("Joining ...");
-            let login_packet = packet::LoginSuccess {
-                uuid: "1c88a2d0-cdf1-4999-9dc1-3e2f696dde05".to_string(),
-                username: packet.username,
-            };
-            println!("{:?}", login_packet);
-            messenger
-                .send(MessengerOperations::Send(SendPacketMessage {
-                    conn_id,
-                    packet: Packet::LoginSuccess(login_packet),
-                }))
-                .unwrap();
-            *state = 3; // Switches to play state
-            println!("state={}", *state);
-            game_state::spawn_player(conn_id, messenger);
+        Packet::LoginStart(login_start) => {
+            *state = 3;
+            confirm_login(p, conn_id, messenger, login_start);
         }
         _ => {
             println!("Login failed");
         }
     }
 }
+
+fn confirm_login(
+    p: Packet,
+    conn_id: i32,
+    messenger: Sender<MessengerOperations>,
+    login_start: packet::LoginStart,
+) {
```

Comment:
this method is giant and could be refactored but I think it'll change a bit once it's informed by gamestate anyway so I'm not going to bother refactoring this temporary code